cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/PushCommandTest.java

```diff
@@ -205,6 +209,59 @@ public void testPushWithCustomPath() throws IOException {
 
     }
 
+    @Test(description = "Push a tool to local repository")
+    public void testPushToolToLocal() throws IOException {
+        Path validBalProject = Paths.get("build").resolve("tool-gayals");
+
+        FileUtils.copyDirectory(
+                this.testResources.resolve("tool-gayals").toFile(), validBalProject.toFile());
+        FileUtils.moveDirectory(
+                validBalProject.resolve("target-dir").toFile(), validBalProject.resolve("custom").toFile());
+
+        Path customTargetDirBalaPath = validBalProject.resolve("custom").resolve("bala")
+                .resolve("gayaldassanayake-tool_gayal-java17-1.1.0.bala");
+        PushCommand pushCommand = new PushCommand(validBalProject, printStream, printStream, false,
+                customTargetDirBalaPath);
+        String[] args = { "--repository=local" };
+        new CommandLine(pushCommand).parse(args);
+
+        Path mockRepo = Paths.get("build").resolve("ballerina-home");
+
+        try (MockedStatic<RepoUtils> repoUtils = Mockito.mockStatic(RepoUtils.class)) {
+            repoUtils.when(RepoUtils::createAndGetHomeReposPath).thenReturn(mockRepo);
+            repoUtils.when(RepoUtils::getBallerinaShortVersion).thenReturn("1.0.0");
+            repoUtils.when(RepoUtils::readSettings).thenReturn(Settings.from());
+            pushCommand.execute();
+        }
+
+        String buildLog = readOutput(true);
+        String actual = buildLog.replaceAll("\r", "");
+        String expected = "Successfully pushed " + customTargetDirBalaPath.toString() + " to 'local' repository.";
+        Assert.assertTrue(actual.contains(expected));
+
+        try {
+            ProjectFiles.validateBalaProjectPath(mockRepo.resolve("repositories").resolve("local")
+                    .resolve("bala").resolve("gayaldassanayake").resolve("tool_gayal")
+                    .resolve("1.1.0").resolve("java17"));
+        } catch (ProjectException e) {
+            Assert.fail(e.getMessage());
+        }
+
+        Path localToolJsonPath = mockRepo.resolve("repositories").resolve("local").resolve("bala")
+                .resolve("local-tools.json");
+
+        Assert.assertTrue(Files.exists(localToolJsonPath));
+
+        try (BufferedReader bufferedReader = Files.newBufferedReader(localToolJsonPath, StandardCharsets.UTF_8)) {
+            JsonObject localToolJson = new Gson().fromJson(bufferedReader, JsonObject.class);
+            JsonObject pkgDesc = localToolJson.get("luhee").getAsJsonObject();
+
+            Assert.assertEquals(pkgDesc.get("org").getAsString(), "gayaldassanayake");
+            Assert.assertEquals(pkgDesc.get("name").getAsString(), "tool_gayal");
+        }
+
+    }
+
```

Comment:
Noticed quite a few new lines added unnecessarily. Would be better check and remove them.